tmt/steps/execute/internal.py

```diff
@@ -244,9 +263,8 @@ def go(self, guest):
 
         self._run_tests(guest)
 
-    def _run_tests(self, guest, extra_environment=None):
+    def _run_tests(self, guest: Guest, extra_environment: EnvironmentType = {}) -> None:
```

Comment:
Never ever use mutable defaults for keyword arguments.

https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/